frame_utils.go

```diff
@@ -0,0 +1,75 @@
+package tchannel
+
+import (
+	"fmt"
+	"runtime"
+	"strings"
+	"sync"
+)
+
+type RecordingFramePool struct {
```

Comment:
(I think this is the reason why I skipped this in the past -- was trying to avoid exporting this type)

Missing godoc, can you add comments for all exported symbols, and also explicitly call out that it's only intended for tests.